src/internal/connector/onedrive/collection.go

```diff
@@ -97,6 +100,8 @@ type Collection struct {
 	// Specifies if it new, moved/rename or deleted
 	state data.CollectionState
 
+	kind collectionKind
```

Comment:
per prior discussion on naming, and a want to follow [these naming conventions](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md), `kind` implies a the body serialization structure, which isn't entirely appropriate for this case.   `type collectionType` would be more appropriate.  Or, noting the consts, maybe `scope collectionScope` would be better?

Also, this could use a header for explanation of its purpose.

Reply:
Ahh, you are right. I wanted to use `type` as well, but since that is a keyword, don't think we will be able to use it. Let me switch to `scope`.